bundles/mapping/mapmodule/MapModuleClass.ol.js

```diff
@@ -530,32 +531,173 @@ export class MapModule extends AbstractMapModule {
-        this.getMap().getView().setCenter([lonlat.lon, lonlat.lat]);
+
         if (zoom === null || zoom === undefined) {
-            zoom = this.getMapZoom();
+            zoom = { type: 'zoom', value: this.getMapZoom() };
+        }
+        if (zoom === Number) {
+            zoom = { type: 'zoom', value: zoom };
         }
-        this.getMap().getView().setZoom(zoom);
+        const zoomValue = zoom.type === 'scale' ? view.getZoomForResolution(zoom.value) : zoom.value;
+
+        this._animateTo(lonlat, zoomValue, animation, duration);
+
         this.updateDomain();
         if (suppressEnd !== true) {
-            this.notifyMoveEnd();
+            setTimeout(() => {
+                this.notifyMoveEnd();
+            }, notifyDuration);
         }
         return true;
     }
+
+    /**
+     * @method tourMap
+     * Moves the map from point to point
+     * @param {Object[]} coordinates array of coordinates to move the map along
+     * @param {Object | Number} zoom absolute zoomlevel to set the map to
+     * @param {Object} options options, such as animation and duration
+     *     Usable animations: fly/pan/zoomPan
+     */
+    tourMap (coordinates, zoom, options) {
+        const view = this.getMap().getView();
+        const me = this;
+        const duration = options && options.duration ? options.duration : 3000;
+        const delayOption = options && options.delay ? options.delay : 750;
+        const animation = options && options.animation ? options.animation : '';
+
+        if (zoom === null || zoom === undefined) {
+            zoom = { type: 'zoom', value: this.getMapZoom() };
+        }
+
+        const zoomDefault = zoom.type === 'scale' ? view.getZoomForResolution(zoom.value) : zoom.value;
+        let index = -1;
+        let delay = 0;
+        let status = { steps: coordinates.length };
+
+        const next = (more) => {
+            if (more) {
+                ++index;
+                if (index < coordinates.length) {
+                    // Check if location has overrides for animation values
+                    const location = coordinates[index];
+                    const zoomValue = location.zoom || zoomDefault;
+                    const animationValue = location.animation || animation;
+                    const durationValue = location.duration || duration;
+                    status = { ...status, step: index + 1 };
+
+                    setTimeout(function () {
+                        me._animateTo(location, zoomValue, animationValue, durationValue, next);
+                        setTimeout(() => me.notifyTourEvent(status), durationValue);
+                    }, delay);
+
+                    // set Delay for next point
+                    delay = location.delay || delayOption;
+                } else {
+                    me.notifyTourEvent(status);
+                }
+            } else {
+                me.notifyTourEvent(status, true);
+            }
+        };
+        next(true);
```

Comment:
Could this be written like this. So always send an event excluding the first call (index is -1). Then we wouldn't have to have so many elses and wouldn't have to set a timeout inside another timeout.

```suggestion
        const next = (more) => {
            if (index !== -1) {
                // trigger if this wasn't the first call
                me.notifyTourEvent(status, !more);
            }
            if (!more) {
                // if we are done we can stop here
                return;
            }
            // go to the next step
            ++index;
            if (index < coordinates.length) {
                // Check if location has overrides for animation values
                const location = coordinates[index];
                const zoomValue = location.zoom || zoomDefault;
                const animationValue = location.animation || animation;
                const durationValue = location.duration || duration;
                status = { ...status, step: index + 1 };

                setTimeout(function () {
                    me._animateTo(location, zoomValue, animationValue, durationValue, next);
                }, delay);

                // set Delay for next point
                delay = location.delay || delayOption;
            }
        };
        next(true);
```

Reply:
Yeah this so much more beautiful than what it was before. I wondered if it would be logical to send status to event on first call also, but have the step be 0 for example?

Reply:
Like a "tour starting" event. Sure why not.